Gems/Atom/RHI/Code/Source/Platform/iOS/Atom_RHI_Traits_iOS.h

```diff
@@ -0,0 +1,8 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+#pragma once
```

Comment:
I think this will fail to compile because Factory.cpp uses the define without checking if it's defined.
I think added `#define AZ_TRAIT_RENDERDOC_MODULE nullptr` should work.

Reply:
See `Factory.cpp:45` which disables the code you're referring to I believe. I left it empty actually because I would want attempting to reference `AZ_TRAIT_RENDERDOC_MODULE` at all on platforms that don't support RenderDoc to fail to compile.

The `USE_RENDERDOC` define is provided in CMake only if a renderdoc shared library or dll is provided at configuration time (and no such binary exists for either iOS or Mac).